src/config.rs

```diff
@@ -0,0 +1,1235 @@
+                Element::Limit => {
+                    eprintln!("warning: busd does not implement `<limit>`");
+                }
+                Element::Listen(s) => {
+                    bc.listen.insert(s);
+                }
+                Element::Pidfile(p) => bc.pidfile = Some(p),
+                Element::Policy(pe) => {
+                    if let Some(p) = OptionalPolicy::try_from(pe)? {
+                        bc.policies.push(p);
+                    }
+                }
+                Element::Servicedir(p) => {
+                    bc.servicedir.push(p);
+                }
+                Element::Servicehelper(p) => {
+                    // NOTE: we're assuming this has the same "last one wins" behaviour as `<type>`
+                    // TODO: warn and then ignore if we aren't reading:
+                    // /usr/share/dbus-1/system.conf
+                    bc.servicehelper = Some(p);
+                }
+                Element::StandardSessionServicedirs => {
+                    // NOTE: because servicedir sequence is important, we cannot de-duplicate this
+                    // TODO: warn and then ignore if we aren't reading: /etc/dbus-1/session.conf
+                    // TODO: include XDG_...-based directories
+                    bc.servicedir
+                        .push(PathBuf::from("/usr/share/dbus-1/services"));
+                }
+                Element::StandardSystemServicedirs => {
+                    // NOTE: because servicedir sequence is important, we cannot de-duplicate this
+                    // TODO: warn and then ignore if we aren't reading:
+                    // /usr/share/dbus-1/system.conf
+                    bc.servicedir
+                        .extend(STANDARD_SYSTEM_SERVICEDIRS.iter().map(PathBuf::from));
+                }
+                Element::Syslog => bc.syslog = true,
+                Element::Type(TypeElement { r#type: value }) => bc.r#type = Some(value),
+                Element::User(s) => bc.user = Some(s),
+            }
+        }
+
+        Ok(bc)
+    }
+}
+
+impl BusConfig {
+    pub fn limits(&self) -> Limits {
+        match self.r#type {
+            Some(BusType::Session) => Limits {
+                /* For the session bus, override the default relatively-low limits
+                with essentially infinite limits, since the bus is just running
+                as the user anyway, using up bus resources is not something we need
+                to worry about. In some cases, we do set the limits lower than
+                "all available memory" if exceeding the limit is almost certainly a bug,
+                having the bus enforce a limit is nicer than a huge memory leak. But the
+                intent is that these limits should never be hit. */
+                // dbus-daemon / dbus-broker is limited to the highest positive number in i32,
+                // but we use u32 here, so we can pick the preferred 4GB memory limits
```

Comment:
* Let's not use `/* */` comments, especially mixing them in the same place.
* nitpick: The line limit is 100 chars, please make use of the full space available to reduce lines. :)